webapp/components/wpt-prs.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright 2018 The WPT Dashboard Project. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+import '../node_modules/@polymer/iron-collapse/iron-collapse.js';
+import '../node_modules/@polymer/paper-button/paper-button.js';
+import '../node_modules/@polymer/polymer/lib/elements/dom-if.js';
+import '../node_modules/@polymer/polymer/lib/elements/dom-repeat.js';
+import { ensureTrailingSlash } from './utils.js';
+import { html } from '../node_modules/@polymer/polymer/lib/utils/html-tag.js';
+import { PolymerElement } from '../node_modules/@polymer/polymer/polymer-element.js';
+import { LoadingState } from './loading-state.js';
+
+class WPTPR extends PolymerElement {
+  static get template() {
+    return html`
+    <style>
+      .time {
+        font-size: 0.8em;
+        font-style: italic;
+      }
+      .pr {
+        display: flex;
+        align-items: center;
+        margin-bottom: 4px;
+      }
+      .pr img {
+        margin-right: 16px;
+        height: 24px;
+        width: 24px;
+      }
+    </style>
+    <div class="pr">
+      <img src="/static/github.svg">
+      <div>
+        <a href="[[pr.html_url]]">#[[pr.number]]</a>: [[pr.title]]
+        <br>
+        <span class="time">
+          opened [[timeAgo(pr.created_at)]] by
+          <a href="[[pr.user.html_url]]">[[pr.user.login]]</a>
+        </span>
+      </div>
+    </div>
+`;
+  }
+
+  static get is() {
+    return 'wpt-pr';
+  }
+
+  static get properties() {
+    return {
+      pr: Object,
+    };
+  }
+
+  timeAgo(time) {
+    const date = new Date(time);
+    const s = Math.floor((new Date() - date) / 1000);
+    const units = [
+      [60 * 60 * 24 * 365, 'years'],
+      [60 * 60 * 24 * 28, 'months'],
+      [60 * 60 * 24 * 7, 'weeks'],
+      [60 * 60 * 24, 'days'],
+      [60 * 60, 'hours'],
+      [60, 'minutes'],
+    ];
+    for (const unit of units) {
+      const scalar = Math.floor(s / unit[0]);
+      if (scalar > 1) {
```

Comment:
Ah, so given the `Math.floor`, we'll say "13 days" and "23 months" before we say "2 weeks" and "2 years". I does avoid the need to pluralize and is fine, just double checking that you intended this.

Reply:
Yep, intended.